### PR TITLE
docs: move vim.health documentation to lua.txt

### PR DESCRIPTION
**Pull request by dundargoc**
`vim.health` is not a "plugin" but part of our Lua API and the
documentation should reflect that. This also helps make the
documentation maintenance easier as it is now generated.

### CODE REVIEW

**Review by clason — APPROVED** [review]
💯

File: runtime/doc/lua.txt [hunk 1]
```diff
@@ -4430,4 +4430,129 @@ tohtml.tohtml({winid}, {opt})                         *tohtml.tohtml.tohtml()*
...
+with healthchecks for configuration, performance, python support, ruby
+support, clipboard support, and more.
+
+To run all healthchecks, use: >vim
+
+        :checkhealth
+<
+Plugin authors are encouraged to write new healthchecks. |health-dev|
+
+Commands                                                     *health-commands*
```
Comment (wookayin) [inline runtime/doc/lua.txt:4448]: nit: the helptag *health* is gone, it might be a good idea to add here.

Comment (dundargoc) [reply runtime/doc/lua.txt:4448]: Added to https://github.com/neovim/neovim/pull/28837

**Comment by justinmk** [comment]
I don't have a complete answer for when modules deserve their own help file, but this one could stay as its own file. There is no requirement that all Lua modules live in lua.txt (though that's good as a _starting point_ until we see a clear need to extract to a dedicated help file).

**Pull request merged**